tests/generators/bls/main.py

```diff
@@ -72,101 +78,160 @@ def hash_message(msg: bytes,
             int_to_hex(fq2.coeffs[0], F2Q_COEFF_LEN),
             int_to_hex(fq2.coeffs[1], F2Q_COEFF_LEN),
         ]
-        for fq2 in bls.utils.hash_to_G2(msg, domain)
+        for fq2 in bls.hash_to_curve.hash_to_G2(msg, DST)
     ]
 
 
-def hash_message_compressed(msg: bytes, domain: bytes) -> Tuple[str, str]:
+def hash_message_compressed(msg: bytes) -> Tuple[str, str]:
     """
     Hash message
     Input:
         - Message as bytes32
-        - domain as bytes8
     Output:
         - Message hash as a compressed G2 point
     """
-    z1, z2 = bls.utils.compress_G2(bls.utils.hash_to_G2(msg, domain))
+    z1, z2 = bls.point_compression.compress_G2(bls.hash_to_curve.hash_to_G2(msg, DST))
     return [int_to_hex(z1, G2_COMPRESSED_Z_LEN), int_to_hex(z2, G2_COMPRESSED_Z_LEN)]
 
 
-def case01_message_hash_G2_uncompressed():
-    for msg in MESSAGES:
-        for domain in DOMAINS:
-            yield f'uncom_g2_hash_{encode_hex(msg)}_{encode_hex(domain)}', {
+def case01_sign():
+    for privkey in PRIVKEYS:
+        for message in MESSAGES:
+            sig = bls.G2ProofOfPossession.Sign(privkey, message)
+            full_name = f'{int_to_hex(privkey)}_{encode_hex(message)}'
+            yield f'sign_case_{(hash(bytes(full_name, "utf-8"))[:8]).hex()}', {
                 'input': {
-                    'message': encode_hex(msg),
-                    'domain': encode_hex(domain),
+                    'privkey': int_to_hex(privkey),
+                    'message': encode_hex(message),
                 },
-                'output': hash_message(msg, domain)
+                'output': encode_hex(sig)
             }
 
 
-def case02_message_hash_G2_compressed():
-    for msg in MESSAGES:
-        for domain in DOMAINS:
-            yield f'com_g2_hash_{encode_hex(msg)}_{encode_hex(domain)}', {
+def case02_verify():
+    for i, privkey in enumerate(PRIVKEYS):
+        for message in MESSAGES:
+            # Valid signature
+            signature = bls.G2ProofOfPossession.Sign(privkey, message)
+            pubkey = bls.G2Basic.PrivToPub(privkey)
```

Comment:
```suggestion
            pubkey = bls. G2ProofOfPossession.PrivToPub(privkey)
```

IMO, it's better to be consistent with the cipher suite we are using.